### DISCUSSION TITLE
Error  while loading a local map on Android

### DISCUSSION THREAD

**Discussion by red0bear** [mapsforge/mapsforge#1363] [Q&A] [answered]
Hello people, i need some help. 

I want load map with rendertheme but he instant loading the thing but it dont download or  find rendertheme

this error appear W/DirectRenderer: Exception: java.lang.IllegalArgumentException: File error for XML rendertheme 

here my actual config on android :

android {
    compileSdk 31

    defaultConfig {
        applicationId ""
        minSdk 21
        targetSdk 31
        versionCode 1
        versionName "1.0"
        multiDexEnabled true
        testInstrumentationRunner "androidx.test.runner.AndroidJUnitRunner"
    }

    buildTypes {
        release {
            minifyEnabled false
            proguardFiles getDefaultProguardFile('proguard-android-optimize.txt'), 'proguard-rules.pro'
        }
    }
    compileOptions {
        sourceCompatibility JavaVersion.VERSION_1_8
        targetCompatibility JavaVersion.VERSION_1_8
    }
}

dependencies {

    implementation 'org.jetbrains.kotlin:kotlin-stdlib-jdk7:1.6.0'
    implementation 'com.android.support:multidex:1.0.3'
   // implementation 'androidx.appcompat:appcompat:1.5.1'

    implementation 'com.google.android.material:material:1.6.0'
    implementation 'androidx.constraintlayout:constraintlayout:2.1.4'
    implementation 'com.google.android.gms:play-services-maps:18.1.0'
    testImplementation 'junit:junit:4.13.2'
    androidTestImplementation 'androidx.test.ext:junit:1.1.3'
    androidTestImplementation 'androidx.test.espresso:espresso-core:3.4.0'
    implementation 'com.android.support:support-annotations:28.0.0'

    implementation 'org.mapsforge:mapsforge-map-android:0.18.0'
    implementation 'org.mapsforge:mapsforge-map:0.18.0'
    implementation 'org.mapsforge:mapsforge-themes:0.18.0'
    implementation 'org.mapsforge:sqlite-android:0.18.0'
    implementation 'com.caverock:androidsvg:1.4'

    implementation 'org.osmdroid:osmdroid-android:6.1.14'
    implementation 'org.osmdroid:osmdroid-mapsforge:6.1.14'
    implementation 'org.osmdroid:osmdroid-wms:6.1.14'
    implementation 'org.osmdroid:osmdroid-shape:6.1.14'

}

i try my best and changed versions to it work but nothing happens. Please people can you help me ? 

exampĺe --> https://stackoverflow.com/questions/42905273/osmdroid-loading-map-file-downloaded-from-mapsforge

**Comment by devemux86** [comment, marked as answer]
For Mapsforge see the [GettingStarted](https://github.com/mapsforge/mapsforge/blob/master/mapsforge-samples-android/src/main/java/org/mapsforge/samples/android/GettingStarted.java) example and the [mapsforge-samples-android](https://github.com/mapsforge/mapsforge/tree/master/mapsforge-samples-android).

For Osmdroid better ask in their forum.

**Reply by red0bear** [reply]
deve i already knoow how to work with stuff on desktop and android , what is werid is theme doesnt load the thing. anyway thank you to answer me.

**Comment by red0bear** [comment]
i open a another issue to see if i can resolve this one for sure. I already try many ways to process tiles from .map extension but seems something is really missing or not downloaded. 

Reference stuff --> https://www.thejavaprogrammer.com/java-lang-illegalargumentexception/
